stable/redis-ha/templates/redis-server-statefulset.yaml

```diff
@@ -1,5 +1,21 @@
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "redis-ha.fullname" . }}-server-svc
+  labels:
+    chart: {{ template "chartref" . }}
+{{ include "labels.standard" . | indent 4 }}
+  annotations:
+{{ toYaml .Values.servers.annotations | indent 4 }}
+spec:
+  selector:
+    redis-node: "true"
+{{ include "labels.standard" . | indent 4 }}
+  type: "{{ .Values.servers.serviceType }}"
```

Comment:
Governing services for statefulsets should be headless, thus type should be hard-coded to `ClusterIP`.